src/main/java/com/microsoft/cse/helium/app/dao/ActorsDao.java

```diff
@@ -11,25 +12,78 @@
 import com.microsoft.azure.spring.data.cosmosdb.core.convert.ObjectMapperFactory;
 import com.microsoft.cse.helium.app.models.Actor;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
 import reactor.core.publisher.Flux;
-// import reactor.core.publisher.Mono;
+import reactor.core.publisher.Mono;
+
+import javax.xml.ws.soap.AddressingFeature;
+import java.util.regex.Pattern;
+
 
 @Service
 public class ActorsDao extends BaseCosmosDbDao {
 
+    private static final Logger _logger = LoggerFactory.getLogger(ActorsDao.class);
+
     final String _actorSelect = "select m.id, m.partitionKey, m.actorId, m.type, m.name, m.birthYear, m.deathYear, m.profession, m.textSearch, m.movies from m where m.type = 'Actor' ";
     final String _actorContains = " and contains(m.textSearch, \"%s\") ";
     final String _actorOrderBy = " order by m.name ";
     final String _actorOffset = " offset %d limit %d ";
 
+    final String _actorSelectById = _actorSelect + " and m.actorId = '%s'";
+
+
     @Autowired
     public ActorsDao(IConfigurationService configService){
         super(configService);
     }
 
+    private Boolean validActorId(String _actorId){
+        /* Valid input: starts with 'nm' (case sensitive)
+        followed by 5-9 digits
+        parses to int > 0
+        */
+        System.out.println(_actorId);
+        Pattern p = Pattern.compile("[nm]{2}[0-9]{5,9}");
+        return p.matcher(_actorId).matches();
+
+    }
+
+
+    public Mono<Actor> getActorById(String _actorId)  {
+        final String query = String.format(_actorSelectById, _actorId.toString());
+
+        if(!validActorId(_actorId)) {
+            _logger.error("Invalid actorId parameter");
+        }
+
+        ObjectMapper objMapper = ObjectMapperFactory.getObjectMapper();
+
+        String readItemQuery = _actorSelectById + _actorId;
+        System.out.println("readItemQuery " + query);
+
+        Mono<Actor> actor =
+                this._context
+                .getBean(CosmosClient.class)
+                .getDatabase(this._cosmosDatabase)
+                .getContainer(this._cosmosContainer)
+                .queryItems(query, this._feedOptions)
+                .flatMap(cosmosItemFeedResponse -> Mono.justOrEmpty(cosmosItemFeedResponse
+                        .results()
+                        .stream()
+                        .map(cosmosItem -> cosmosItem.toObject(Actor.class))
+                        .findFirst()))
+                .onErrorResume(Mono::error)
+                .next();
+        return actor;
+
```

Comment:
You may as well just return this._context. No need for the actor variable. Also, unnecessary blank line

Reply:
done!